src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java

```diff
@@ -149,9 +149,17 @@ public ProvisionConfiguration updateIdentityConfiguration(ProvisionContext provi
              MqttClientConnection connection = mqttConnectionHelper
                 .getMqttConnection(mqttParameterBuilder.clientBootstrap(clientBootstrap).build())) {
 
-            CompletableFuture<Boolean> connected = connection.connect();
-            FutureExceptionHandler.getFutureAfterCompletion(connected,
-                    "Caught exception while establishing connection to AWS Iot");
+            while (true) {
+                try {
+                    CompletableFuture<Boolean> connected = connection.connect();
+                    FutureExceptionHandler.getFutureAfterCompletion(connected,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:** 
The loop containing this operation may iterate indefinitely if the operation throws an exception on multiple instances of its evaluation. As the operation throws an exception, it bypasses exit points in the loop and continues to the next iteration through a catch clause.

**Fix:** 
Add a retry counter to ensure that the loop terminates after a predefined number of iterations.